applications/OptimizationApplication/tests/algorithm_tests/test_convergence.py

```diff
@@ -60,10 +60,106 @@ def test_L2(self):
         self.optimization_problem.AdvanceStep()
         algorithm_data.GetBufferedData()["search_direction"] = search_direction / 100000
         self.assertTrue(convergence_criterium.IsConverged())
+
+    def test_L2_max(self):
+        param = Kratos.Parameters("""{
+            "type"              : "l2_norm",
+            "max_iter"          : 2,
+            "tolerance"         : 1e-4
+        }""")
+        algorithm_data = ComponentDataView("algorithm", self.optimization_problem)
+        convergence_criterium = CreateConvergenceCriteria(param, self.optimization_problem)
+        search_direction = KratosOA.CollectiveExpression([Kratos.Expression.ElementExpression(self.model_part)])
+        KratosOA.CollectiveExpressionIO.Read(search_direction, KratosOA.CollectiveExpressionIO.PropertiesVariable(Kratos.DENSITY))
+        algorithm_data.GetBufferedData()["search_direction"] = search_direction
+        self.assertFalse(convergence_criterium.IsConverged())
+        self.optimization_problem.AdvanceStep()
+        algorithm_data.GetBufferedData()["search_direction"] = search_direction
+        self.assertFalse(convergence_criterium.IsConverged())
         self.optimization_problem.AdvanceStep()
         algorithm_data.GetBufferedData()["search_direction"] = search_direction
         self.assertTrue(convergence_criterium.IsConverged())
```

Comment:
I would add a test for `max_iter` as well

Reply:
It is a test for max iter